weather_dl/download_pipeline/parsers.py

```diff
@@ -328,8 +328,9 @@ def require(condition: bool, message: str) -> None:
     num_partition_keys = len(partition_keys)
     if use_date_as_directory(config):
         num_partition_keys -= 1
-        if str(params.get('target_path')).endswith('/'):
-            params['target_path'] = params.get('target_path')[:-1]
+        target_path = t.cast(str, params.get('target_path', ''))
+        if target_path != '':
```

Comment:
One last nit: It this if statement necessary? I think `rstrip` on an empty string will produce an empty string.

Reply:
Right, but I added it because you don't want to add a key to the dict if it doesn't already have it.

Reply:
Though, generally, I'm not actually sure why this is located here, because it seems like it'd be more natural to handle it in prepare_target_name().

On a similar note, I think subtracting 1 from num_partition_keys here, but actually removing the "date" partition key in prepare_target_name(), might be a little risky. Could they be done in the same place?

Reply:
> Right, but I added it because you don't want to add a key to the dict if it doesn't already have it.
Ah, that makes sense. 

> it seems like it'd be more natural to handle it in prepare_target_name().
I agree – this would be a better fit: It'd be best if the parser had less logic, in general. 

> Could they be done in the same place?
Yes, that totally sounds right to me. 

Last: I think we are no longer using the `num_partition_keys` counter anymore. It would be prudent to pull that out.